plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,181 @@
+import io.trino.spi.function.SchemaFunctionName;
+import io.trino.spi.ptf.AbstractConnectorTableFunction;
+import io.trino.spi.ptf.Argument;
+import io.trino.spi.ptf.Descriptor;
+import io.trino.spi.ptf.ScalarArgument;
+import io.trino.spi.ptf.ScalarArgumentSpecification;
+import io.trino.spi.ptf.TableFunctionAnalysis;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.base.util.Functions.checkFunctionArgument;
+import static io.trino.plugin.deltalake.DeltaLakeColumnType.SYNTHESIZED;
+import static io.trino.plugin.deltalake.DeltaLakeErrorCode.DELTA_LAKE_BAD_DATA;
+import static io.trino.plugin.deltalake.DeltaLakeErrorCode.DELTA_LAKE_INVALID_TABLE;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogDir;
+import static io.trino.plugin.deltalake.transactionlog.checkpoint.TransactionLogTail.getEntriesFromJson;
+import static io.trino.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.trino.spi.ptf.ReturnTypeSpecification.GenericTable.GENERIC_TABLE;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class TableChangesFunction
+        extends AbstractConnectorTableFunction
+{
+    private static final String SCHEMA_NAME = "system";
+    private static final String NAME = "table_changes";
+    public static final SchemaFunctionName TABLE_CHANGES_NAME = new SchemaFunctionName(SCHEMA_NAME, NAME);
+    public static final String SCHEMA_NAME_ARGUMENT = "SCHEMA_NAME";
+    private static final String TABLE_NAME_ARGUMENT = "TABLE_NAME";
+    private static final String SINCE_VERSION_ARGUMENT = "SINCE_VERSION";
+    private static final String CHANGE_TYPE_COLUMN_NAME = "_change_type";
+    private static final String COMMIT_VERSION_COLUMN_NAME = "_commit_version";
+    private static final String COMMIT_TIMESTAMP_COLUMN_NAME = "_commit_timestamp";
+
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final DeltaLakeMetadataFactory deltaLakeMetadataFactory;
+
+    public TableChangesFunction(TrinoFileSystemFactory fileSystemFactory, DeltaLakeMetadataFactory deltaLakeMetadataFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(SCHEMA_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(SINCE_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build()),
```

Comment:
What does "since_version = -1" mean (in the case when the argument is not provided and takes the default value? If we want to make it optional to represent "since the beginning of time", then it should just be optional with no default value.

Reply:
@homar, @findepi, this comment is still unaddressed

Reply:
Sorry I somehow missed that comment. 
So this is supposed to be a trick not to do deal with null value.
Someone else, @findinpath or @alexjo2144 or both also thought that null would be better so I will switch to it. @findepi do you agree?

Reply:
the versions start at 0
the since_version is exclusive. we debated this and decided this is the best thing to do due to how this will be used. equivalent of `git log start_version..`)

when user doesn't specify since_version, we provide "since the beginning of time"
when user specifies since_version= -1, this means the same

i think @martint your concern is about using -1 as a special value. we're not doing that. -1 isn't any special.

@homar please don't change here.

Reply:
What do you mean by “it’s not special”? Since version -1 doesn’t exist, the -1 value *is* special.

Reply:
that first version is 0 and since_version is exclusive, so -1  is the natural way to show changes committed in version 0.
(version 0 may or may not have rows)

Reply:
The natural way is to omit the `since_version` argument if you want to see the changes since the beginning of time.

Reply:
Just stumbled across this conversation when looking at the docs as well and wondering how this table version stuff works. From looking at the Delta Lake docs from Databricks and Microsoft (who would have thought ... ) I find nothing about a valid version value of -1 or anything below zero for that matter. It seems to start with zero and go up. 

I did not see any example of a query where you can retrieve a -1 version of a table and get a result. Not sure if this is worth testing and even if it works in Delta Lake .. might be a bug. Is there a spec for Delta Lake that could be consulted?

Inventing a new value of -1 as valid in Trino seems to be a bad idea. As such I think we need to change this. 

However I am also not that deep in the Trino or Delta Lake codebase to I will let you decide together. Ultimately the decision lies with @martint as the language lead to ensure consistency in Trino overall and with the ecosystems we collaborate with. 

https://docs.delta.io/latest/versioning.html
https://learn.microsoft.com/en-us/azure/databricks/delta/history
https://trino.io/development/roles.html#language-lead